### PR TITLE
Implement support for using multiple fonts

### PR DESCRIPTION
**Pull request by mzn723**
Implement support for using multiple fonts.
Updating raqm-test.
Adding a new test file to test multiple fonts.

### CODE REVIEW

**Comment by khaledhosny** [comment]
Please add a test where font change happens in the middle of an Arabic word.

**Comment by mzn723** [comment]
Test added.

The tested text is " عربي" where, the middle two letters "ر,ب" are in a different font.

**Comment by khaledhosny** [comment]
We need a way to communicate the font used for each glyph back to the caller, so I think `raqm_glyph_t` should be extended with a pointer to the face used.

**Comment by mzn723** [comment]
Adding FT_Face to raqm_glyphs_t

**Comment by khaledhosny** [comment]
The Travis build is failing, please check it.

**Comment by khaledhosny** [comment]
Please merge the last three commits into the first one, and reorder the commits so that adding the explicit font option _precedes_ fixing the multi font support.

File: tests/runtest [hunk 1]
```diff
@@ -27,7 +27,27 @@ do
...
+			do
+				if [ x"$fonts" = x"" ]
+				then
+					fonts=$srcdir/$f
+				else
+					fonts=$fonts,$srcdir/$f
+				fi
+			done
+			IFS=$oldIFS
+			$testtool --text="$text" $opts --fonts="$srcdir/$fonts" > $output
```
Comment (khaledhosny) [inline tests/runtest]: Adding `$srddir` here is wrong since we added it for each font already. This makes running the tests in out-of-tree builds fail.

**Comment by mzn723** [comment]
On 8 Feb 2016 8:56 am, "Khaled Hosny" <notifications@github.com> wrote:
>
> Please merge the last three commits into the first one, and reorder the
commits so that adding the explicit font option precedes fixing the multi
font support.
>
Just to make sure that I understood this correctly,  total commits now are
4,  I should combine the last three into one commit, so the total commits
will be 2 with the explicit font commit as the first in order (as it is
now) ?

**Comment by khaledhosny** [comment]
Yes.

**Comment by mzn723** [comment]
Updated.
 $srddir  fixed and the latest 3 commits were merged into one.

**Comment by khaledhosny** [comment]
Travis build is still failing.

**Comment by mzn723** [comment]
FAIL: gtkdoc-check.test what does this has to do with my commits?!

**Comment by khaledhosny** [comment]
Master builds fine, so you need to debug it.

**Pull request merged**